### ISSUE TITLE
Let's have a chat between different forks? :)

### ISSUE THREAD

**Issue by rusefillc** [provideyourown/gerbmerge#7]
https://github.com/rusefi/gerbmerge has a dozen fixes related to KiCAD 6, 'rotate' fixes and at least one new feature 'flip' all done by @andreika-git

https://github.com/frank-f/gerbmerge is the one with most latest change @frank-f please comment on this ticket if you notice this ticket

https://github.com/ckemere/gerbmerge/  last commit Mar 7, 2021 @ckemere please comment on this ticket if you notice this ticket

https://github.com/jnicolson/gerbmerge/ last commit 2019 @jnicolson please comment on this ticket if you notice this ticket

**Comment by ckemere** [comment]
I'd love to know what the latest and greatest repo was! In particular, Python 3 would be awesome...

**Comment by jnicolson** [comment]
Hi - my fork should work on Python 3 (@ckemere), this was the main reason i worked on it.  It was a while ago but it looks like i also refactored some things (split gerber and excellon parsing into different files for example)

**Comment by frank-f** [comment]
Well, hi 😄
My fork contains a few changes and fixes I made by myself, but also a few that I found while looking through other forks. It'd be nice to see those integrated into a collaboration.

> I'd love to know what the latest and greatest repo was! In particular, Python 3 would be awesome...

I had a similar idea and took a short look at that, but that'll be a much larger undertaking. I haven't used too many features of gerbmerge myself yet, but a Python 3 migration would require exhaustive testing. So I put that idea aside for now.

**Comment by rusefillc** [comment]
@frank-f do you want to enable "issues" on your fork by the way? Same question for @jnicolson and @ckemere :)

Python 3 https://github.com/provideyourown/gerbmerge/issues/8

Let's invite @sakoPO @eyal0 @GIPdA @bobc @provideyourown @olegart into this party

Question: shall we attempt anything as a group of people interested in gerbmerge? What can we attempt?

Shall we attempt a round of upstream PRs? Shall we go ask around asking to remove some of the abandoned forks? Is there nothing to do here really?

**Comment by frank-f** [comment]
> Hi - my fork should work on Python 3 (@ckemere), this was the main reason i worked on it. It was a while ago but it looks like i also refactored some things (split gerber and excellon parsing into different files for example)

Nice work on the Python 3 support. I looked through your repo briefly and I think you have put the most work into your fork. If we're not finding some upstream to combine all our commits, I vote we continue based on your fork.

**Comment by rusefillc** [comment]
@frank-f I am really glad that you are proposing https://github.com/jnicolson/gerbmerge as the new 'primary' fork!!! I've already created a couple of tickets there :)

**Comment by eyal0** [comment]
My changes were minor, fixing bugs and adding support for G85.  I reviewed my additions just now and I think that they were done well.  If you like, please merge eyal0/master branch into whatever branch you like.

If you have questions, let me know.

**Comment by jnicolson** [comment]
> @frank-f I am really glad that you are proposing https://github.com/jnicolson/gerbmerge as the new 'primary' fork!!! I've already created a couple of tickets there :)

I'm happy to help where i can - i'm not actively using gerbmerge any more (Even after the changes i made there was still too much i wanted to do) but i'm happy to do bits and pieces if it helps others - i'll probably just need others to test them.

**Comment by frank-f** [comment]
Did one of you manage to merge KiCad 6 gerbers yet? I'm getting lots of strange errors with Aperture Macros. Looking at older projects, it doesn't seem KiCad produced these before at all, so I don't even know what the script could expect as valid data.

**Comment by rusefillc** [comment]
yes https://github.com/rusefi/gerbmerge/tree/rusefi-tests branch contains https://github.com/rusefi/gerbmerge/tree/rusefi-tests/examples/hellen121vag-c which are pretty recent KiCAD 5.99 gerbers - both inputs and merged outputs

**Comment by frank-f** [comment]
I looked through the recent commits, but could not find anything that would solve my issue.

So I went looking, when this was added to KiCad and found [KiCad issue #2020](https://gitlab.com/kicad/code/kicad/-/issues/2020). Changes seem to have been committed for 6.0.0-rc1.

Gerbmerge even contains a note about these macros:
`Define and manage aperture macros (%AM command). Currently,
only macros without replaceable parameters (e.g., $1, $2, etc.)
are supported.`

So as of now I would say, gerbmerge is incompatible with KiCad versions 6.0.0-rc1 onwards, since every rectangular pad seems to have rounded corners and I don't see any option to disable this.

**Comment by frank-f** [comment]
Turns out, KiCad has an option to disable the use of aperture macros. It says "not recommended", but adding support for it to gerbmerge seems to be a larger task since they're very versatile.

**Comment by rusefillc** [comment]
FYI @andreika-git has ported https://github.com/rusefi/gerbmerge to python 3

see https://github.com/rusefi/gerbmerge/commit/f78db83293e9737d4ee4e0d5d377e908ba9aba91

**Comment by kemerelab** [comment]
I just posted an issue - setup.py is still Python 2.0 unless I missed something?

**Comment by andreika-git** [comment]
> I just posted an issue - setup.py is still Python 2.0 unless I missed something?

Thanks for the report! Done:

https://github.com/rusefi/gerbmerge/commit/43eb62470565f5f408cf8094047624bc43f90183
